src/road_cmd.cpp

```diff
@@ -2040,6 +2040,14 @@ static TrackStatus GetTileTrackStatus_Road(TileIndex tile, TransportType mode, u
 
 					trackdirbits = TrackBitsToTrackdirBits(AxisToTrackBits(axis));
 					if (IsCrossingBarred(tile)) red_signals = trackdirbits;
+					if (IsLevelCrossingTile(TileAddByDiagDir(tile, AxisToDiagDir(axis))) &&
+							IsCrossingBarred(TileAddByDiagDir(tile, AxisToDiagDir(axis)))) {
+						red_signals &= (TrackdirBits)0x0102; // magic value. I think TRACKBIT_X_SW and TRACKBIT_X_NE should be swapped
```

Comment:
"I think" always gives me the idea you knew what you were doing ;) Any way to make this "I think" into a sure thing, without us diving deep into this code?

Reply:
honest answer? i wrote that 10 years ago. i've no clue what i meant either :p

Reply:
So, i reviewed this, and the expression evaluates to
`red_signals &= TRACKDIR_BIT_Y_SE | TRACKDIR_BIT_X_SW;`
and i think the comment meant that i should switch that out with
`red_signals &= ~(TRACKDIR_BIT_Y_NW | TRACKDIR_BIT_X_NE);`
which should be functionally identical, as the diagonal TrackDirBits can never be set.